client.go

```diff
@@ -131,6 +134,18 @@ func WithPointerClass(pointerClass string) ClientOption {
 	}
 }
 
+// WithObjectPrefix attach a prefix to the object key (prefix/uuid)
+func WithObjectPrefix(prefix string) ClientOption {
+	r, _ := regexp.Compile("^[0-9a-zA-Z!_.*'()-]+$")
```

Comment:
You could avoid compiling the regex every time by exposing it as package level variable. Also, you can probably avoid the need to check for errors (since you're pretty certain the regex is fine) by using `MustCompile` instead.

```go
var  validObjectNameRegex = regexp.MustCompile(`^[0-9a-zA-Z!_.*'()-]+$`)

func WithObjectPrefix(prefix string) ClientOptions {
   //...
}
```

Reply:
True, I'm moving the regex definition to the top of the file 😀